sdk/nodejs/policy/server.ts

```diff
@@ -819,3 +804,45 @@ function pushNotApplicable(
     notApplicable.push(na);
     return notApplicable;
 }
+
+/**
+ * Helper for creating a reportViolation function.
+ */
+function makeReportViolationFun(
+    policyName: string,
+    policyPackName: string,
+    policyPackVersion: string,
+    description: string,
+    enforcementLevel: EnforcementLevel,
+    ds: Diagnostic[],
+): ReportViolation {
+    return (message, urn?) => {
+        if (typeof message === "object") {
+            const policyInfo: ReportViolationArgs = message;
+            ds.push({
+                policyName: policyInfo.name || policyName,
+                policyPackName,
+                policyPackVersion,
+                message: policyInfo.message,
+                urn: policyInfo.urn,
+                description: policyInfo.description || policyInfo.message,
```

Comment:
I'm wondering if this should be:

```suggestion
                description: policyInfo.description || description,
```